optin/puller/workflows/generate_postoptin.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import asyncio
+import json
+from binascii import unhexlify
+from pathlib import Path
+
+from symbolchain.CryptoTypes import PublicKey
+from symbolchain.sc import TransactionType
+from symbolchain.symbol.Network import Address
+
+from puller.client.SymbolClient import SymbolClient
+
+OPTIN_SIGNER_PUBLIC_KEYS = '7AEC08AA66CB50B0C3D180DE7508D5D82ECEDCDC9E73F61FA7069868BEABA856'
+
+
+def parse_args():
+	parser = argparse.ArgumentParser(description='generate postoptin json')
+	parser.add_argument('--symbol-node', help='Symbol node url', default='http://wolf.importance.jp:3000')
+	parser.add_argument('--optin-signer-public-keys', help='optin signer public keys (comma separated)', default=OPTIN_SIGNER_PUBLIC_KEYS)
+	parser.add_argument('--output', help='output path', default='post-optin.json')
+	return parser.parse_args()
+
+
+def to_sym_address(hex_string):
+	return Address(unhexlify(hex_string))
+
+
+class RedeemProcessor:
+	def __init__(self, client):
+		self.client = client
+		self.redemptions = []
+
+	async def process(self, public_key):
+		start_id = None
+		while True:
+			transactions = await self.client.redeem_transactions(public_key, start_id)
+			if not transactions:
+				break
+
+			for transaction in transactions:
+				self.process_transaction(transaction)
+
+			start_id = transactions[-1]['id']
+			print('.', end='', flush=True)
+
+	def process_transaction(self, transaction_meta):
+		transaction = transaction_meta['transaction']
+		if TransactionType.TRANSFER.value != transaction['type']:
+			return
+
+		# note: ignoring txes without message
+
+		if 'message' in transaction:
+			message = unhexlify(transaction['message']).decode('utf8')
+			assert message[0] == '\0'
+
+			# "new" format
```

Comment:
hm?

Reply:
'new' format is the one in json `{'nisaddress':'...'}`